logic/src/commonMain/kotlin/com/wire/kalium/logic/data/event/EventRepository.kt

```diff
@@ -104,26 +101,31 @@ class EventDataSource(
         currentClientId().flatMap { clientId -> liveEventsFlow(clientId) }
 
     private suspend fun liveEventsFlow(clientId: ClientId): Either<NetworkFailure, Flow<WebSocketEvent<EventEnvelope>>> =
-        wrapApiRequest { notificationApi.listenToLiveEvents(clientId.value) }.map {
-            it.map { webSocketEvent ->
-                when (webSocketEvent) {
-                    is WebSocketEvent.Open -> {
-                        flowOf(WebSocketEvent.Open())
-                    }
-
-                    is WebSocketEvent.NonBinaryPayloadReceived -> {
-                        flowOf(WebSocketEvent.NonBinaryPayloadReceived<EventEnvelope>(webSocketEvent.payload))
-                    }
-
-                    is WebSocketEvent.Close -> {
-                        flowOf(WebSocketEvent.Close(webSocketEvent.cause))
-                    }
-
-                    is WebSocketEvent.BinaryPayloadReceived -> {
-                        eventMapper.fromDTO(webSocketEvent.payload, true).asFlow().map { WebSocketEvent.BinaryPayloadReceived(it) }
+        wrapApiRequest { notificationApi.listenToLiveEvents(clientId.value) }.map { webSocketEventFlow ->
+            flow {
+                webSocketEventFlow.collect { webSocketEvent ->
+                    when (webSocketEvent) {
```

Comment:
Why using a flow builder and collect data inside it and not just map the flow as we currently do ?

Reply:
Reverted it and added `flatMapConcat` on top of function